data_steward/validation/ehr_merge.py

```diff
@@ -0,0 +1,189 @@
+            # elif field_name.endswith('_id') and not field_name.endswith('concept_id') and field_type == 'integer':
+            # not using this because we refer to other ids in some tables which will get overwritten
+            elif field_name == table_name + '_id':
+                col_expr = 'ROW_NUMBER() OVER() as %(field_name)s ' % locals()
+            else:
+                col_expr = field_name
+            col_exprs.append(col_expr)
+        col_expr_str = ',\n  '.join(col_exprs)
+        q = 'SELECT\n  '
+        q += ',\n  '.join(col_exprs)
+        q += '\nFROM'
+        q += '\n ('
+        q_blocks = []
+        for hpo in hpos_to_merge:
+            q_dum = ' ( SELECT * FROM `%(project_id)s.%(dataset_id)s.%(hpo)s_%(table_name)s` t' % locals()
+            if visit_id_flag and hpo in hpos_with_visit:
+                q_dum += '\n LEFT JOIN  '
+                q_dum += VISIT_ID_MAPPING_TABLE_SUBQUERY % locals()
+            q_dum += ')'
+            q_blocks.append(q_dum)
+        if len(q_blocks) == 0:
+            return ""
+        q += "\n UNION ALL \n".join(q_blocks)
+        q += ')'
+        return q
+
+
+def query(q, destination_table_id, write_disposition):
+    """
+    Run query, write to stdout any errors encountered
+    :param q: SQL statement
+    :param destination_table_id: if set, output is saved in a table with the specified id
+    :param write_disposition: WRITE_TRUNCATE, WRITE_APPEND or WRITE_EMPTY (default)
+    :return: query result
+    """
+    qr = bq_utils.query(q, destination_table_id=destination_table_id, write_disposition=write_disposition)
+    if 'errors' in qr['status']:
+        logging.error('== ERROR ==')
+        logging.error(str(qr))
+    return qr
+
+
+def create_mapping_table(hpos_with_visit, project_id, dataset_id):
+    # # list of hpos with visit table and creating visit id mapping table queries
+    visit_hpo_queries = []
+    for hpo in hpos_with_visit:
+        visit_hpo_queries.append(VISIT_ID_HPO_BLOCK % locals())
+    union_all_blocks = '\n UNION ALL'.join(visit_hpo_queries)
+    visit_mapping_query = VIST_ID_MAPPING_QUERY_SKELETON % locals()
+    logging.info('Loading ' + VISIT_ID_MAPPING_TABLE)
+    query_result = query(visit_mapping_query,
+                         destination_table_id=VISIT_ID_MAPPING_TABLE,
+                         write_disposition='WRITE_TRUNCATE')
+    visit_mapping_query_job_id = query_result['jobReference']['jobId']
+    incomplete_jobs = bq_utils.wait_on_jobs([visit_mapping_query_job_id], retry_count=10)
+    if len(incomplete_jobs) == 0:
+        query_result = bq_utils.get_job_details(visit_mapping_query_job_id)
+        if 'errors' in query_result['status']:
+            logging.error('{} load failed!'.format(VISIT_ID_MAPPING_TABLE))
+            return "visit mapping table failed"
+    else:
+        logging.error('{} load taking too long!'.format(VISIT_ID_MAPPING_TABLE))
+        raise RuntimeError("visit mapping table failed")
```

Comment:
Modify runtime error message to say load is taking too long like in the logging. Please include the job id as well.